CK.Core/ByteArrayWriter.cs

```diff
@@ -0,0 +1,224 @@
+using System;
+using System.Buffers;
+using System.Diagnostics;
+
+namespace CK.Core
+{
+    /// <summary>
+    /// Fast and direct implementation of a <see cref="IBufferWriter{T}"/> of bytes
+    /// that exposes its <see cref="Buffer"/> and can change its current <see cref="Position"/>.
+    /// </summary>
+    /// <remarks>
+    /// This is based on the <see cref="ArrayBufferWriter{T}"/>.
+    /// </remarks>
+    public sealed class ByteArrayWriter : IBufferWriter<byte>
```

Comment:
Totally. There's one variation that could be implemented and useful: instead of allocating its array, a BytePoolArrayWriter would use the ArrayPool<byte> and reallocates/returns the array to the pool. It would hence need to be IDisposable (to return the last allocated array).